bridge/engine/Engine.cpp

```diff
@@ -35,6 +35,16 @@ Engine::Engine(jobmanager::JobManager& backgroundJobQueue)
     }
 }
 
+Engine::Engine(jobmanager::JobManager& backgroundJobQueue, std::thread&& outsideThread)
+    : _messageListener(nullptr),
+      _running(true),
+      _tickCounter(0),
+      _tasks(1024),
+      _thread(std::move(outsideThread))
+{
+    // This construct is for unit test so it does not initialize a thread
```

Comment:
Are there tests where this is used??

Reply:
Yes, My new unit tests on API level `ApiRequestHandlerTest.cpp` It starts a MixerManager with an `Engine` with an empty thread `std::thread`. It is a bit complex. So we start an `MixerManagerSpy` (which extend from `MixerManager` to have access to the `protected` fields) and that class intialize an `Engine` with an empty thread

https://github.com/finos/SymphonyMediaBridge/blob/96300d2e1fc5edd18907e7c2d29012e8ff69416b/test/include/mocks/MixerManagerSpy.h#L62